server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -187,9 +201,26 @@ void execute() {
                 try (var ignored = threadPool.getThreadContext().clearTraceContext()) {
                     fetchIndicesStats();
                 }
+                try (var ignored = threadPool.getThreadContext().clearTraceContext()) {
+                    fetchNodesHeapUsage();
+                }
             }
         }
 
+        private void fetchNodesHeapUsage() {
+            heapUsageSupplier.getClusterHeapUsage(ActionListener.releaseAfter(new ActionListener<>() {
+                @Override
+                public void onResponse(Map<String, HeapUsage> stringHeapUsageMap) {
```

Comment:
Naming nit:
```suggestion
                public void onResponse(Map<String, HeapUsage> currentHeapUsageMap) {
```

Reply:
Done in e26b62fcd76ab8ab03d11c7f3cb15cf34e7c72bf